feature-libs/commerce-quotes/components/commerce-quotes-actions-by-role/commerce-quotes-actions-by-role.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component } from '@angular/core';
+import {
+  CommerceQuotesFacade,
+  QuoteAction,
+  QuoteState,
+} from '@spartacus/commerce-quotes/root';
+import { ConfigurationService } from '@spartacus/core';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-commerce-quotes-actions-by-role',
+  templateUrl: './commerce-quotes-actions-by-role.component.html',
+})
+export class CommerceQuotesActionsByRoleComponent {
+  quoteDetailsState$ = this.commerceQuotesService.getQuoteDetails().pipe(
+    map((detailsState) => {
+      const { data } = detailsState;
+      if (data) {
+        const { allowedActions, state } = data;
+        if (!allowedActions || !state) {
+          return detailsState;
+        }
+        return {
+          ...detailsState,
+          data: {
+            ...data,
+            allowedActions: this.getOrderedList(state, allowedActions),
+          },
+        };
+      }
+    })
```

Comment:
If we're using some model mapping here along with config for feature, I think we should create a normalizer for such a case. This way we provide a clear and readable way of transforming/mapping backend response into our local model type.

Reply:
Side note: maybe it's worth double checking the above comment with Nik / Kris about whether this case requires a normalizer.

Reply:
Good Idea! Going further, we could add an indicator to each action to have information if it's primary or not, instead of another filtering in component. I've asked Nik and I'm waiting for his thoughts about this :)

Reply:
Logic has been moved to normalizer with separate unit tests. Component is now clean and shiny 👌